modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/CurrencySnapshotConsensusFunctions.scala

```diff
@@ -56,7 +56,11 @@ object CurrencySnapshotConsensusFunctions {
     ): F[Unit] =
       stateChannelSnapshotService.consume(signedArtifact, context) >>
         gossipForkInfo(gossip, signedArtifact) >>
-        maybeDataApplication.traverse_(_.onSnapshotConsensusResult(signedArtifact))
+        maybeDataApplication.traverse_ { da =>
+          signedArtifact.toHashed.flatMap { hashedArtifact =>
+            da.onSnapshotConsensusResult(hashedArtifact)
+          }
+        }
```

Comment:
```suggestion
        maybeDataApplication.traverse_(da => signedArtifact.toHashed >>= da.onSnapshotConsensusResult)
```

Reply:
or using `flatMap` it would be:
```scala
signedArtifact.toHashed.flatMap(da.onSnapshotConsensusResult)
```

up to you @scasplte2

Reply:
thank you for the suggestions, I've updated to use the `>>=` since similar operators are used earlier in the function